.github/workflows/near_crates_publish.yml

```diff
@@ -0,0 +1,32 @@
+name: near crates publish
+
+on:
+  push:
+    branches: master
+  workflow_dispatch:
+
+jobs:
+  publish-cargo-crates:
+    name: "Publish near-workspaces on crates.io https://crates.io/crates/near-workspaces"
+    runs-on: "ubuntu-latest"
+    environment: deploy
+    permissions:
+      contents: write # required for crates push
+    timeout-minutes: 30 
+    
+    steps:
+      - name: Publish near-workspaces on crates.io
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+        run: |
+          set -x
+          cargo install --git https://github.com/miraclx/cargo-workspaces --tag v0.3.0 cargo-workspaces
```

Comment:
Use a caching action like the CI does to install binaries.

Reply:
if I understand correctly the action doc (Swatinem/rust-cache), this cache is only useful for multiple steps within same workflow/job.
In this case we only do it one time and never reuse later

Reply:
I was meaning to suggest https://github.com/baptiste0928/cargo-install rather than `Swatinem/rust-cache` but in either case the cached files are saved and reused between workflow runs. You can see the list of cached files [here](https://github.com/near/nearcore/actions/caches).